AsyncUDP missing when using ESP32 board

With a platform set to esp32, you get the following error:

```
In file included from src/esphome.h:22,
                 from src/main.cpp:3:
src/esphome/components/espsense/espsense.h:12:10: fatal error: AsyncUDP.h: No such file or directory

******************************************************************
* Looking for AsyncUDP.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:AsyncUDP.h"
* Web  > https://registry.platformio.org/search?q=header:AsyncUDP.h
*
******************************************************************

 #include "AsyncUDP.h"
          ^~~~~~~~~~~~
compilation terminated.
*** [.pioenvs/geo-panel/src/main.cpp.o] Error 1
```

--- a/components/espsense/__init__.py
+++ b/components/espsense/__init__.py
@@ -56,6 +56,8 @@ CONFIG_SCHEMA = (
 async def to_code(config):
     if CORE.is_esp8266:
         cg.add_library("ESPAsyncUDP", "")
+    elif CORE.is_esp32:
+        cg.add_library("ESP32 Async UDP", None)
 
     var = cg.new_Pvariable(config[CONF_ID])
     await cg.register_component(var, config)

